backend/assignments/1/src/main/java/org/example/CoinsDetails.java

```diff
@@ -0,0 +1,33 @@
+package org.example;
+
+import org.slf4j.LoggerFactory;
+import java.util.HashMap;
+import java.util.List;
+public class CoinsDetails {
+    public static final HashMap<String, Coins> symbolToCoins = Main.symbolToCoins;
+    private static CoinsDetails coinInfo = new CoinsDetails();
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(CoinsDetails.class);
+    private CoinsDetails(){
+
+    }
+
+    public static CoinsDetails getCoinInfo() {
+        if(coinInfo == null){
+            coinInfo = new CoinsDetails();
+        }
+        return coinInfo;
+    }
+
+    public static synchronized Coins getCoins(String address) {
```

Comment:
There is no use in making `getCoins()` function synchronous. Try to make the coins array itself inside a synchronous.